### PR TITLE
Mixer parsing fixes

### LINKED ISSUE #5118: Mixer loading corner case
If we try to load the following mixer it seems to get sent to IO but one of the last 2 mixers doesn't get loaded.

```
R: 2- 10000 10000 10000 0
Z:
M: 2
O:      10000  10000      0 -10000  10000
S: 1 0  10000  10000      0 -10000  10000
S: 1 1  10000  10000      0 -10000  10000
M: 2
O:       10000   10000    0 -10000  10000
S: 1 0   10000   10000    0 -10000  10000
S: 1 1  -10000  -10000    0 -10000  10000
```

It seems to split it up at a bad location during transfer. If I extend it a bit (add another null mixer before), or remove one of the minus signs in the first servo mixer, or change `PX4IO_P_CONFIG_MAX_TRANSFER` from 64 to 32, the mixer loads correctly.

### PR DESCRIPTION
**Pull request by LorenzMeier**
Fixed a parsing corner case and validated in unit test that the fix holds up independent of the packetization interval.

To run in one command:

```sh
make tests
```

To run from the shell:

```sh
make posit_sitl_shell none
pxh> tests mixer
```

Fixes #5118

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Yeah:

```sh
______  __   __    ___ 
| ___ \ \ \ / /   /   |
| |_/ /  \ V /   / /| |
|  __/   /   \  / /_| |
| |     / /^\ \ \___  |
\_|     \/   \/     |_/

px4 starting.

pxh> tests mixer
INFO  [tests] RUNNING TEST: loadIOPass
INFO  [tests] TEST PASSED: loadIOPass
INFO  [tests] RUNNING TEST: loadQuadTest
INFO  [tests] TEST PASSED: loadQuadTest
INFO  [tests] RUNNING TEST: loadVTOL1Test
INFO  [tests] TEST PASSED: loadVTOL1Test
INFO  [tests] RUNNING TEST: loadVTOL2Test
INFO  [tests] TEST PASSED: loadVTOL2Test
INFO  [tests] RUNNING TEST: mixerTest
INFO  [tests] TEST PASSED: mixerTest
INFO  [modules__unit_test] ALL TESTS PASSED
INFO  [modules__unit_test]   Tests passed :      5
INFO  [modules__unit_test]   Tests failed :      0
INFO  [modules__unit_test]   Tested assertions : 810
pxh> 
```

@AndreasAntener Can you try with the hardware with this branch?

**Comment by AndreasAntener** [comment]
Great! I verified this on my bench Pixhawk. I also changed the vtol2 mixer slightly to have the combination that failed for me before (I checked again which one it was exactly). Ran the mixer test on hardware as well.

File: ROMFS/px4fmu_test/mixers/complex_test.mix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+M: 1
```
Comment (AndreasAntener) [inline ROMFS/px4fmu_test/mixers/complex_test.mix]: O-oh, loading fail when this is set to 5

Comment (AndreasAntener) [inline ROMFS/px4fmu_test/mixers/complex_test.mix]:
```
INFO  [tests] RUNNING TEST: loadComplexTest
ERROR [tests] Mixer text length overflow
ERROR [tests] Mixer load failed with chunk size 16
```

Comment (LorenzMeier) [inline ROMFS/px4fmu_test/mixers/complex_test.mix]: Stupid me. I'll fix. Can you test your airframe meanwhile?

Comment (AndreasAntener) [inline ROMFS/px4fmu_test/mixers/complex_test.mix]: Home office today, later this week.

**Comment by LorenzMeier** [comment]
@AndreasAntener I got a little crazy and I'm now actually linking the test to the IO buffer sizes and I'm loading *ALL* mixers. That means you now can't contribute a mixer which will not load OK.

And it found corner cases which I fixed with the next commits:

```
pxh> tests mixer
INFO  [tests] RUNNING TEST: loadIOPass
INFO  [tests] TEST PASSED: loadIOPass
INFO  [tests] RUNNING TEST: loadQuadTest
INFO  [tests] TEST PASSED: loadQuadTest
INFO  [tests] RUNNING TEST: loadVTOL1Test
INFO  [tests] TEST PASSED: loadVTOL1Test
INFO  [tests] RUNNING TEST: loadVTOL2Test
INFO  [tests] TEST PASSED: loadVTOL2Test
INFO  [tests] RUNNING TEST: loadComplexTest
ERROR [tests] Mixer text length overflow for file: ROMFS/px4fmu_test/mixers/complex_test.mix. Is PX4IO_MAX_MIXER_LENGHT too small? (curr len: 220)
ERROR [tests] Mixer load failed with chunk size 55
ERROR [tests] TEST FAILED: loadComplexTest
INFO  [tests] RUNNING TEST: loadAllTest
INFO  [tests] Testing all mixers in ROMFS/px4fmu_common/mixers/
ERROR [tests] Mixer text length overflow for file: ROMFS/px4fmu_common/mixers/blade130.main.mix. Is PX4IO_MAX_MIXER_LENGHT too small? (curr len: 220)
ERROR [tests] Mixer load failed with chunk size 7
ERROR [tests] Error testing mixer ROMFS/px4fmu_common/mixers/blade130.main.mix
ERROR [tests] TEST FAILED: loadAllTest
INFO  [tests] RUNNING TEST: mixerTest
INFO  [tests] TEST PASSED: mixerTest
ERROR [modules__unit_test] SOME TESTS FAILED
INFO  [modules__unit_test]   Tests passed :      5
INFO  [modules__unit_test]   Tests failed :      2
INFO  [modules__unit_test]   Tested assertions : 1390
Command 'tests' failed, returned -1
```

**Comment by LorenzMeier** [comment]
Here is the output of the fixed test:

```

______  __   __    ___ 
| ___ \ \ \ / /   /   |
| |_/ /  \ V /   / /| |
|  __/   /   \  / /_| |
| |     / /^\ \ \___  |
\_|     \/   \/     |_/

px4 starting.

pxh> tests mixer
INFO  [tests] RUNNING TEST: loadIOPass
INFO  [tests] TEST PASSED: loadIOPass
INFO  [tests] RUNNING TEST: loadQuadTest
INFO  [tests] TEST PASSED: loadQuadTest
INFO  [tests] RUNNING TEST: loadVTOL1Test
INFO  [tests] TEST PASSED: loadVTOL1Test
INFO  [tests] RUNNING TEST: loadVTOL2Test
INFO  [tests] TEST PASSED: loadVTOL2Test
INFO  [tests] RUNNING TEST: loadComplexTest
INFO  [tests] TEST PASSED: loadComplexTest
INFO  [tests] RUNNING TEST: loadAllTest
INFO  [tests] Testing all mixers in ROMFS/px4fmu_common/mixers/
INFO  [tests] TEST PASSED: loadAllTest
INFO  [tests] RUNNING TEST: mixerTest
INFO  [tests] TEST PASSED: mixerTest
INFO  [modules__unit_test] ALL TESTS PASSED
INFO  [modules__unit_test]   Tests passed :      7
INFO  [modules__unit_test]   Tests failed :      0
INFO  [modules__unit_test]   Tested assertions : 4248
pxh> 
```

**Comment by LorenzMeier** [comment]
@AndreasAntener Please review / test on HW and merge if it passes.

**Comment by AndreasAntener** [comment]
Merged via #6245

**Pull request closed without merging**